### PR TITLE
String concat

### PR DESCRIPTION
**Pull request by kbrock**
Didn't really produce the savings I was expecting.
Basically, it is the same thing.

So don't accept this request.

But it is an alternative view of the problem.

So wanted to forward along.

### CODE REVIEW

File: lib/stamp/emitters/two_digit.rb [hunk 1]
```diff
@@ -13,9 +13,9 @@ def initialize(field, &block)
         @modifier = block
       end
 
-      def format(target)
+      def format(out, target)
         value = modify(target.send(field))
-        '%2.2d' % value
+        out << (value < 10 ? '0' : '') << value.to_s
```
Comment (kbrock) [inline lib/stamp/emitters/two_digit.rb]: yea, this should be:
```ruby
out << '0' if value < 10
out << value
```

**Comment by kbrock** [comment]
@jeremyw any thoughts of this?

Funny. I was just about to suggest this PR again.
I did notice that the `join` code was converted to an `each` and `<<`. So most of the speed improvement is probably already realized.

Locally my raw benchmarks seem to suggest this could get 10% speed improvement. Mostly around the `out << '0'` use case.

If you like this idea, I'll rebase and play. Otherwise, I'll let live and probably stumble upon this in another year or so.

File: lib/stamp/emitters/composite.rb [hunk 2]
```diff
@@ -7,12 +7,9 @@ def initialize
         @emitters = []
       end
 
-      def format(target)
-        # NOTE using #each to build string because benchmarking shows
-        # that it's ~20% faster than .map.join('')
-        result = ''
-        @emitters.each { |e| result << e.format(target).to_s }
-        result
+      def format(out, target)
```
Comment (jeremyw) [inline lib/stamp/emitters/composite.rb]: I'm not sure why this is better. It seems like it just distributes string-building repsonsibility to the emitters.

Comment (kbrock) [inline lib/stamp/emitters/composite.rb]: yes. that is all it is.
In the cases where you want to output more than 1 string, it ends up being quicker to `lambda |result, target| result << str1 << str2 }` than `result << lambda { |target| str1 + str2 }`.

Granted it is probably 10% in trivial benchmarks, but hey. that is something right?
(well, if you think it is much uglier, than that probably doesn't warrant the change)

It is hard to benchmark across these changes. I like benchmark-ips, but not sure how to do it across sweeping changes like this. Only can do it with lambdas.

**Pull request closed without merging**